docs/DeveloperGuide.md

```diff
@@ -341,6 +341,70 @@ without breaking existing syntax coloration managed with `TextMate` in `IntelliJ
 </extensions>
 ```
 
+## Language Server API
+
+If you need to `enable/disable` and / or `start/stop` your language server, LSP4IJ provides the LanguageServerManager API which provides this support:
+
+### Enable / Disable 
+
+By default, language server is enabled : in other words, when IntelliJ IDEA starts LSP4IJ tracks the opened files and starts the proper language servers.
+If you need to manage this enable state, you can do that by implementing in your factory [LanguageServerEnabledSupport](https://github.com/redhat-developer/lsp4ij/blob/main/src/main/java/com/redhat/devtools/lsp4ij/LanguageServerEnabledSupport.java):
+
+```java
+package my.language.server;
+
+import com.intellij.openapi.project.Project;
+import com.redhat.devtools.lsp4ij.LanguageServerFactory;
+import com.redhat.devtools.lsp4ij.LanguageServerEnabledSupport;
+import com.redhat.devtools.lsp4ij.client.LanguageClientImpl;
+import com.redhat.devtools.lsp4ij.server.StreamConnectionProvider;
+import org.jetbrains.annotations.NotNull;
+
+public class MyLanguageServerFactory implements LanguageServerFactory, LanguageServerEnabledSupport {
+
+    ...
+  
+    @Override
+    public boolean isEnabled(@NotNull Project project) {
+        // Get enabled state from your settings
+      boolean enabled = ...
+      return enabled;
+    }
+
+    @Override
+    public void setEnabled(boolean enabled, Project project) { {
+      // Update enabled state of your settings 
+    }
+  
+}
+```
+
+To enable, disable your language server you can use the LanguageServerManager API:
+
+```java
+Project project = ...
+boolean enabled = // get enabled state from your settings
+LanguageServerManager.getInstance(project).setEnable("myLanguageServerId", enabled);
+```
+
+### Start / Stop
+
+To `start` your language server you can use the LanguageServerManager API. 
+
+```java
+Project project = ...
+LanguageServerManager.getInstance(project).start("myLanguageServerId");
+```
+
+The language server will start only, if there is one opened file which matches your language server:

```

Comment:
that might be too restrictive, you can have LS that support working without opened files, e.g. JDT.LS

Reply:
I support now StartOptions#forceStart which start the language server in any case.